src/components/search.js

```diff
@@ -1,60 +1,81 @@
 import React from 'react';
-
+import {AsyncTypeahead} from 'react-bootstrap-typeahead';
 import Navbar from 'react-bootstrap/lib/Navbar';
-import FormGroup from 'react-bootstrap/lib/FormGroup';
-import FormControl from 'react-bootstrap/lib/FormControl';
-import InputGroup from 'react-bootstrap/lib/InputGroup';
-import Button from 'react-bootstrap/lib/Button';
-import Glyphicon from 'react-bootstrap/lib/Glyphicon';
 
 export default class Search extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
+      searchSuggestions: [],
       searchQuery: ''
     };
   }
 
-  doSearch(e) {
-    e.preventDefault();
+  doSearch(searchQuery) {
+    if (!searchQuery) {
+      return;
+    }
+    this.setState({ searchQuery });
 
-    const { searchQuery } = this.state;
     const { lat, lon, zoom } = this.props;
 
     // fetch(`https://www.freemap.sk/api/0.1/q/${encodeURIComponent(searchQuery)}&lat=${lat}&lon=${lon}&zoom=${zoom}`, {
     fetch(`https://nominatim.openstreetmap.org/search/${encodeURIComponent(searchQuery)}`
-        + `?format=jsonv2&lat=${lat}&lon=${lon}&zoom=${zoom}&namedetails=1&extratags=1`, {
+        + `?format=jsonv2&lat=${lat}&lon=${lon}&zoom=${zoom}&namedetails=1&extratags=1&countrycodes=SK`, {
       method: 'GET'
     }).then(res => res.json()).then(data => {
-      const searchResults = data.map((d, id) => ({ id, lat: d.lat, lon: d.lon, tags: { name: d.namedetails.name} }));
-      this.props.onSearchResultsUpdate(searchResults);
+      const searchSuggestions = data.map((d, id) => {
+        const name = d.namedetails.name;
+        const tags = { name, type: d.type };
+        return { id, label: name, lat: d.lat, lon: d.lon, tags };
+      });
+      this.setState({ searchSuggestions });
     });
   }
 
-  updateSearchQuery(e) {
-    this.setState({ searchQuery: e.target.value });
+  onSelectionChange(selectedResults) {
+    this.props.onSelectSearchResult(selectedResults[0]);
+  }
+
+  onSuggestionHighlightChange(result) {
+    if (result && result.lat) {
+      result.lat = parseFloat(result.lat);
+    }
+    if (result && result.lon) {
+      result.lon = parseFloat(result.lon);
+    }
```

Comment:
tu som pridal parsovanie lebo api vracia string (si mohol vidiet warningy v js konzoly). ale dam to na spravnejsie miesto :-)